src/Symfony/Component/Process/Process.php

```diff
@@ -429,12 +429,31 @@ public function wait(?callable $callback = null): int
         do {
             $this->checkTimeout();
             $running = $this->isRunning() && ('\\' === \DIRECTORY_SEPARATOR || $this->processPipes->areOpen());
-            $this->readPipes($running, '\\' !== \DIRECTORY_SEPARATOR || !$running);
+            if ($fiber = \Fiber::getCurrent()) {
+                $this->readPipes(false, '\\' !== \DIRECTORY_SEPARATOR || !$running);
+                $startedAt = microtime(true);
+                $fiber->suspend();
+                $sleepFor = (int) (1000 - (microtime(true) - $startedAt) * 1000000);
+                if (0 < $sleepFor) {
```

Comment:
(same below)
```suggestion
                $suspendedAt = microtime(true);
                $fiber->suspend();
                if (0 < $sleepFor = (int) (1000 - (microtime(true) - $suspendedAt) * 1000000)) {
```

But I would also make readPipe return true/false whether the process did yield some activity or not. This way, we won't force a sleep while there are things to do.